src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -8302,7 +8300,149 @@ private ErrorPropertySymbol CreateErrorPropertySymbol(ImmutableArray<PropertySym
             }
         }
 
-        internal bool ReportDelegateInvokeUseSiteDiagnostic(BindingDiagnosticBag diagnostics, TypeSymbol possibleDelegateType,
+#nullable enable
+        internal NamedTypeSymbol? GetMethodGroupDelegateType(BoundMethodGroup node, ref CompoundUseSiteInfo<AssemblySymbol> useSiteInfo)
+        {
+            if (GetUniqueSignatureFromMethodGroup(node) is { } method &&
+                GetMethodGroupOrLambdaDelegateType(method.RefKind, method.ReturnsVoid ? default : method.ReturnTypeWithAnnotations, method.ParameterRefKinds, method.ParameterTypesWithAnnotations, ref useSiteInfo) is { } delegateType)
+            {
+                return delegateType;
+            }
+            return null;
+        }
+
+        /// <summary>
+        /// Returns one of the methods from the method group if all methods in the method group
+        /// have the same signature, ignoring parameter names and custom modifiers. The particular
+        /// method returned is not important since the caller is interested in the signature only.
+        /// </summary>
+        private MethodSymbol? GetUniqueSignatureFromMethodGroup(BoundMethodGroup node)
+        {
+            MethodSymbol? method = null;
+            foreach (var m in node.Methods)
+            {
+                switch (node.ReceiverOpt)
+                {
+                    case BoundTypeExpression:
+                        if (!m.IsStatic) continue;
```

Comment:
Is it possible to reach this `continue`?

Reply:
Yes, for an instance overload when the method group is an explicitly qualifed static overload.
```csharp
abstract class A
{
    internal void F(object x) { }
}
class B : A
{
    internal static void F() { }
    static void Main()
    {
        new B().M();
    }
}
```